lib/generator._js

```diff
@@ -297,54 +297,44 @@ ScriptGenerator.prototype.generateWapDeploymentScript = function (_) {
-      msbuildArguments: msbuildArguments,
-      dotnetpublishArguments: dotnetpublishArguments,
-      restore: nugetRestore,
-      solutionPath: this.solutionPath
-    };
-
-    this.generateAspNetCoreScript('aspnet.core.msbuild.template', options, _);
-  } else {
-    // use dotnet command, not msbuild if NO solutionfile is provided
-    // or if there's a solution file, but preview3 project
-    var nugetOrDotnetRestore = this.projectPath.endsWith(".csproj") ? 'dotnet restore': nugetRestore;
-    if (this.solutionPath) {
-        nugetOrDotnetRestore += ' "' + this.solutionPath + '"';
-    } else {
-        nugetOrDotnetRestore += ' "' + this.projectPath + '"';
-        // current directory does not have .sln or .csproj/.xproj, nuget restore will throw an error
-    }
-    this.generateAspNetCoreScript('aspnet.core.template', {aspNetCoreProject: this.projectPath, restore:nugetOrDotnetRestore}, _);
+  if (this.scriptType != ScriptType.batch && this.scriptType != ScriptType.posh) {
+      // TODO what's the support on linux
+    throw new Error('Only batch and posh script files are supported for ASP.NET Core Application');
   }
+  // this.solutionPath and this.projectPath are both relative
+
+  // restore needs to be performed over the entire solution (multiple projects)
+  var options = {RestoreArguments:((this.solutionPath) ? this.solutionPath:this.projectPath)}
+  if (this.projectPath.endsWith(".xproj")){
+      // template msbuild14
+      // XXX xproj almost always comes with sln
+      options.MSBuildArguments = options.RestoreArguments;
+      options.MSPublishArguments = this.projectPath;
+      this.generateAspNetCoreScript('aspnet.core.msbuild14.template', options, _);
+  }else{
+      // "project.json" or "csproj", we all use dotnet cli
+      options.DotnetpublishArguments = this.projectPath;
+      this.generateAspNetCoreScript('aspnet.core.template', options, _);
+  }
+  // TODO replace above code with the following
+  // if (this.projectPath.endsWith("project.json")){
+  //     // template dotnet
+  //     // XXX preview2- projects created with dotnet cli/ yeoman
+  //     // TODO dotnet restore should run without argument since dependencies in "project.json" works different than "xproj/csproj"
+  //     options.DotnetpublishArguments = this.projectPath;
+  //     this.generateAspNetCoreScript('aspnet.core.template', options, _);
+  // } else if (this.projectPath.endsWith(".xproj")){
+  //     // template msbuild14
+  //     // XXX xproj almost always comes with sln
+  //     options.MSBuildArguments = options.RestoreArguments;
+  //     options.MSPublishArguments = this.projectPath;
+  //     this.generateAspNetCoreScript('aspnet.core.msbuild14.template', options, _);
+  // } else {
+  //     // template msbuild15
+  //     // XXX msbuild15 cannot deal with xproj/project.json only csproj
+  //     options.MSBuildArguments = this.projectPath
+  //     this.generateAspNetCoreScript('aspnet.core.msbuild15.template', options, _);
```

Comment:
you are right, it's possible for a dotnet core project with "csproj" to endup on linux:
```
 else {
     // template msbuild15
     // XXX msbuild15 cannot deal with xproj/project.json only csproj
     options.MSBuildArguments = this.projectPath
     this.generateAspNetCoreScript('aspnet.core.msbuild15.template', options, _);
```
but it looks like we just discovered that msbuild15 worked different than our expectation, this is now on hold